app/src/renderer/components/staking/ModalWithdrawAllRewards.vue

```diff
@@ -0,0 +1,68 @@
+<template>
+  <action-modal
+    id="modal-withdraw-all-rewards"
+    ref="actionModal"
+    :submit-fn="submitForm"
+    :validate="isValid"
+    title="Withdraw"
+    class="modal-withdraw-rewards"
+    submission-error-prefix="Withdrawal failed"
+  >
+    <tm-form-group
+      class="action-modal-form-group"
+      field-id="amount"
+      field-label="Amount"
+    >
+      <span class="input-suffix">{{ bondDenom }}</span>
+      <tm-field
+        id="amount"
+        v-model="totalRewards"
+        type="number"
+        readonly
+      />
+    </tm-form-group>
+  </action-modal>
+</template>
+
+<script>
+import { mapGetters } from "vuex"
+import { atoms } from "../../scripts/num.js"
+import ActionModal from "common/ActionModal"
+import TmField from "common/TmField"
+import TmFormGroup from "common/TmFormGroup"
+
+export default {
+  name: `modal-withdraw-all-rewards`,
+  components: {
+    ActionModal,
+    TmField,
+    TmFormGroup
+  },
+  computed: {
+    ...mapGetters([`bondDenom`, `distribution`]),
+    totalRewards({ bondDenom, distribution } = this) {
+      const rewards = distribution.totalRewards[bondDenom]
+      return (rewards && atoms(rewards)) || 0
+    }
+  },
+  methods: {
+    isValid() {
```

Comment:
what is the purpose of this? why not use vuelidate?

Reply:
it fails otherwise, see https://github.com/cosmos/voyager/pull/2128#pullrequestreview-209253070

Reply:
`ActionModal` requires a `validate` function for the parent form.